server/impl/meetup.go

```diff
@@ -1,15 +1,297 @@
+		return InternalServerError{}
+	}
+
+	return operations.NewPatchMeetupIDOK().WithPayload(dbMeetupToModelMeetup(&dbMeetup, id.(string)))
+
+}
+
+func (i *Implementation) updateDBMeetup(ctx context.Context, dbMeetup *db.Meetup) error {
+	tx := i.DB().WithContext(ctx)
+	return tx.Model(dbMeetup).
+		Select("title, time, description, max_capacity, min_capacity, location_lat, location_lon, location_url, location_name").
+		Updates(dbMeetup).Error
+}
+
+func (i *Implementation) createDBMeetup(ctx context.Context, dbMeetup *db.Meetup) error {
+	tx := i.DB().WithContext(ctx)
+	return tx.Create(dbMeetup).Error
+}
+
+func (i *Implementation) insertTagsIntoDB(ctx context.Context, dbMeetup *db.Meetup, modelMeetup *models.Meetup) error {
+	// Insert tags into the DB
+	tx := i.DB().WithContext(ctx)
+
+	var placeholders []string
+	var variables []interface{}
+	for _, tag := range modelMeetup.Tags {
+		placeholders = append(placeholders, "(CURRENT_TIMESTAMP, CURRENT_TIMESTAMP, ?)")
+		variables = append(variables, tag)
+	}
+
+	variables = append(variables, sql.Named("meetup_id", dbMeetup.ID))
+	var err error
+	if len(modelMeetup.Tags) > 0 {
+		// https://stackoverflow.com/a/42217872/1937836
+		err = tx.Raw(fmt.Sprintf(`
+	WITH
+		input_rows(created_at, updated_at, name) AS (VALUES %s),
+		ins AS (       -- insert any tags not yet in the DB
+			INSERT INTO tags (created_at, updated_at, name)
+			SELECT * FROM input_rows
+			ON CONFLICT (name) DO NOTHING
+			RETURNING created_at, updated_at, deleted_at, id, name
+		),
+		all_tags AS (  -- get IDs of tags already in the DB
+			SELECT * FROM ins
+			UNION ALL
+			SELECT t.created_at, t.updated_at, t.deleted_at, t.id, t.name FROM input_rows
+			JOIN tags t using (name)
+		),
+		ignored1 AS (  -- delete stale tag-meetup associations
+			DELETE FROM meetup_tag tm
+			WHERE tm.meetup_id = @meetup_id AND tm.tag_id NOT IN (SELECT id FROM all_tags)
+		),
+		ignored2 AS (  -- insert new tag-meetup associations, if not already exist
+			INSERT INTO meetup_tag (tag_id, meetup_id)
+			SELECT id, @meetup_id FROM all_tags
+			ON CONFLICT (tag_id, meetup_id) DO NOTHING
+		)
+	SELECT * FROM all_tags
+	`, strings.Join(placeholders, ", ")), variables...).Scan(&dbMeetup.Tags).Error
+	} else {
+		err = tx.Model(dbMeetup).Association("Tags").Clear()
+		dbMeetup.Tags = nil
+	}
+	return err
+}
+
+func (i *Implementation) modelMeetupToDBMeetup(ctx context.Context, dbMeetup *db.Meetup, modelMeetup *models.Meetup) error {
+	// Update tags first. Do it through SQL since GORM Association mode Replace
+	// doesn't work reliably when the "tags" table has a unique name constraint.
+	// https://gorm.io/docs/associations.html#Replace-Associations
+	dbMeetup.Title = modelMeetup.Title
+	dbMeetup.Description = modelMeetup.Description
+	dbMeetup.MaxCapacity = *modelMeetup.MaxCapacity
+	dbMeetup.MinCapacity = *modelMeetup.MinCapacity
+	dbMeetup.Description = modelMeetup.Description
+
+	id, _ := strconv.ParseUint(string(modelMeetup.Owner), 10, 64)
```

Comment:
I think you make a good point, but actually in `modelMeetupRequestBodyToModelMeetup`, we set `modelMeetup.Owner` equal to the ID of the logged-in user, so it's not coming directly form the request body. I just want to avoid having to pass in the id as a separate parameter when its already in the model, but let me know what you think